packages/@storefront/flux-capacitor/src/core/actions/validators.ts

```diff
@@ -181,3 +181,40 @@ export const isNotFetching: Validator<boolean> = {
     !Selectors.infiniteScroll(state).isFetchingBackward,
   msg: 'is already fetching'
 };
+
+export const hasValidLabels: Validator<Actions.Payload.Sort> = {
+  func: ({ labels }) => {
+    return !labels
+      || (
+        Array.isArray(labels)
+        && !!labels.length
+        && labels.every((label) => typeof label === 'string')
```

Comment:
If you use `some`, you won't have to check that the array is not empty. Up to you if you don't think it's clear.

Reply:
I'll revisit this as part of the 'allow empty arrays' feedback (provided during demo).